webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,389 @@
+        style: "margin-top:50px;width:80%",
+        oninput: (e) => {
+          passphase = e.target.value;
+        },
+      }),
+      m(
+        "button",
         {
+          style: "margin-top:160px;",
           onclick: () =>
             rs.rsJsonApiRequest(
-              '/rsIdentity/createIdentity',
+              "/rsIdentity/updateIdentity",
               {
-                name,
-                pseudonimous,
-                pgpPassword,
+                id: v.attrs.details.mId,
+                name: v.attrs.name,
+                pseudonimous: false,
+                pgpPassword: passphase,
               },
-              data => {
+              (data) => {
                 const message = data.retval
-                  ? 'Successfully created identity.'
-                  : 'An error occured while creating identity.';
+                  ? "Successfully created identity."
+                  : "An error occured while creating identity.";
                 widget.popupMessage([
-                  m('h3', 'Create new Identity'),
-                  m('hr'),
+                  m("h3", "Create new Identity"),
+                  m("hr"),
                   message,
                 ]);
-                m.redraw();
-              },
+              }
             ),
         },
-        'Create',
+        "Enter"
       ),
     ],
   };
 };
 
 const EditIdentity = () => {
+  let name = "";
   return {
-    view: () => [
-      m('i.fas.fa-user-edit'),
-      m('h3', 'Edit Identity'),
-      m('hr'),
-      m('input[type=text][placeholder=Name]', {}),
-      m('canvas'),
-      m('button', 'Save'),
+    view: (v) => [
+      m("i.fas.fa-user-edit"),
+      m("h3", "Edit Identity"),
+      m("hr"),
+      m("input[type=text][placeholder=Name]", {
+        value: name,
+        oninput: (e) => {
+          name = e.target.value;
+        },
+      }),
+      m("canvas"),
+      m(
+        "button",
+        {
+          onclick: () => {
+            !people_util.checksudo(v.attrs.details.mPgpId)
+              ? widget.popupMessage([
+                  m(SignedEditIdentity, {
+                    name: name,
+                    details: v.attrs.details,
+                  }),
+                ])
+              : rs.rsJsonApiRequest(
+                  "/rsIdentity/updateIdentity",
+                  {
+                    id: v.attrs.details.mId,
+                    name: name,
+                    //avatar: v.attrs.details.mAvatar.mData.base64,
+                    pseudonimous: true,
+                  },
+                  (data) => {
+                    const message = data.retval
+                      ? "Successfully Updated identity."
+                      : "An error occured while updating  identity.";
+                    widget.popupMessage([
+                      m("h3", "Update Identity"),
+                      m("hr"),
+                      message,
+                    ]);
+                    m.redraw();
```

Comment:
Is redraw required here? It's not used in `SignedEditIdentity`.

Reply:
As I know about m.redraw is used to refresh the Page. So I used it here for refreshing the page whenever we edit the identity.

Reply:
Not exactly refresh, but triggers dom update.

Can you please check if the ids list gets updated without `m.redraw()`?
If it doesn't update, we will have to update signed edit identity component also.

Reply:
I checked it. It does not update the DOM  without m.redraw().

Reply:
Even after using m.redraw () it is not updating the DOM. I already created issue #28 for it. I tried to use [m.set](https://mithril.js.org/route.html#mrouteset) for updating the DOM but it takes me to the Login Page. I need your help.

Reply:
Ah, yes! Redraw will not work because the popup window is not part of the main app mount. I can see two solutions here:
- Move the popup menu component into into the main app mount. We will have to move it soon regardless of this issue, or the popup layer will become hard to maintain separately.
- Create a separate model object to keep track of the ids state. This component has grown quite big, so separate state management should be a good way to proceed.

Thanks for creating the issue, we can track it separately. No need to waste more time on this PR.

Reply:
Thanks, @rottencandy, I will create a PR soon based on Second Solution.